### DISCUSSION TITLE
A blog post about the API?

### DISCUSSION THREAD

**Discussion by imbolc** [trillium-rs/trillium#22] [Q&A]
I'd love to read from you, as an insider, about different approaches to server-side web frameworks in Rust. As a user, I can guess that Rocket gives more compile time guaranties, comparing to eg Actix. Tide and warp seems more straightforward and easy to use for smaller things like microservices. Dropshot and Rocket (potentially) can auto-generate OpenApi specs. Would you elaborate on these trade-offs with maybe some examples of APIs comparing to once taken by Trillium? Maybe in form of a blogpost to have people responses?

**Comment by jbr** [comment]
Hi and thanks! I do intend to write a series of blog posts in the near future about trillium's current design and the roadmap for the future, but probably not through the lens of comparison to other rust frameworks. Trillium is still very young, and is just a small part of what I hope it can grow into

**Comment by imbolc** [comment]
Yeah, as a Rust newbee, I personally feel a lack of such compassion for both frontend and backend frameworks. And it seems like there's only a few people who can do it. Mostly people just say - it's an Elm (Phoenix, ...) idiom, but even if I know what it is, it's hard to assess how good it fits into Rust specificity. Anyway I really like how concise and intuitive Trillium examples feel and looking forward for the blog posts :)

**Comment by jbr** [comment]
> I really like how concise and intuitive Trillium examples feel

Thanks! This is really motivating for me. The ultimate goal is to have an easy and approachable interface to learn and start out that also can scale up to complex applications with serious performance demands
